src/nc_ssl.c

```diff
@@ -266,7 +266,8 @@ ssize_t
 nc_ssl_read(SSL *ssl, void *buf, size_t num) {
     int bytes_read;
     while ((bytes_read = SSL_read(ssl, buf, (int)num)) <= 0) {
-        int read_errno = errno;
```

Comment:
Reading errno doesn’t reset it. As long as there are no other functions called that might set it (mainly sys calls) you should be fine.

Reply:
Ah. Either way, we're down to just one errno read, so no need to have a separate variable.